### PR TITLE
[Agent] Hashmap storage box pointer #21158 #21167

### PR DESCRIPTION
**Pull request by yuanchaoa**
### This PR is for:

- Agent

#### Affected branches
- main
- 6.1
- 6.2
#### Checklist
- [ ] Added unit test to verify the fix.
- [ ] Verified eBPF program runs successfully on linux 4.14.x.
- [ ] Verified eBPF program runs successfully on linux 4.19.x.
- [ ] Verified eBPF program runs successfully on linux 5.2.x.

### CODE REVIEW

File: agent/src/collector/flow_aggr.rs [hunk 1]
```diff
@@ -268,24 +266,13 @@ impl FlowAggr {
...
-        if stash_cap > self.stash_init_capacity {
-            let max_history = self.history_length.iter().fold(0, |acc, n| acc.max(*n));
-            if stash_cap > 2 * max_history {
-                // shrink stash if its capacity is larger than 2 times of the max stash length in the past HISTORY_RECORD_COUNT flushes
-                self.metrics.stash_shrinks.fetch_add(1, Ordering::Relaxed);
-                slot_map.shrink_to(self.stash_init_capacity.max(2 * max_history));
-            }
-        }
-
+        slot_map.shrink_to_fit();
```
Comment (rvql) [inline agent/src/collector/flow_aggr.rs:273]: 每次都 shrink 太频繁

File: agent/src/flow_generator/protocol_logs/mod.rs [hunk 2]
```diff
@@ -307,18 +307,18 @@ impl From<AppProtoLogsBaseInfo> for flow_log::AppProtoLogsBaseInfo {
 
 impl AppProtoLogsBaseInfo {
     // 请求调用回应来合并
-    fn merge(&mut self, log: AppProtoLogsBaseInfo) {
+    fn merge(&mut self, log: &AppProtoLogsBaseInfo) {
```
Comment (rvql) [inline agent/src/flow_generator/protocol_logs/mod.rs:310]: 这里为啥要改成传引用？下面一堆 clone 都是没必要的

File: agent/src/collector/flow_aggr.rs (hunk 1 above)
Comment (yuanchaoa) [reply agent/src/collector/flow_aggr.rs:273]: 分钟级的无所谓吧

File: agent/src/flow_generator/protocol_logs/mod.rs (hunk 2 above)
Comment (yuanchaoa) [reply agent/src/flow_generator/protocol_logs/mod.rs:310]: 聚合的hashmap使用了box，这里只有字符串clone了， 直接传值需要 *Log , 整体copy开销更大

**Review by sharang — DISMISSED** [review]

File: agent/src/flow_generator/protocol_logs/mod.rs (hunk 2 above)
Comment (rvql) [reply agent/src/flow_generator/protocol_logs/mod.rs:310]: 原来全是 move，没有整体 copy。反倒是这里 L#425 加了一个多余的 clone

**Review by sharang — APPROVED** [review]

**Review by rvql — APPROVED** [review]

**Pull request merged**